lib/spack/spack/cmd/repo.py

```diff
@@ -420,6 +445,30 @@ def repo_set(args):
     tty.msg(f"Updated repo '{namespace}'")
 
 
+def repo_update(args: Any) -> int:
+    config = spack.config.CONFIG
+    scope = args.scope
+
+    if args.namespace:
+        existing: Dict[str, Any] = config.get("repos", default={}, scope=scope)
+        if args.namespace not in existing:
+            raise SpackError(f"{args.namespace} is not a known repository namespace.")
+
+        entry = spack.util.path.canonicalize_path(args.namespace)
+        descriptors = {
+            args.namespace: spack.repo.parse_config_descriptor(
+                args.namespace or "<unnamed>", entry, lock=spack.repo.package_repository_lock()
+            )
+        }
```

Comment:
@haampie there must be a better way to do this I didn't immediately see.

Reply:
I think you can just run

```python
descriptors = spack.repo.RepoDescriptors.from_config(
            spack.repo.package_repository_lock(), spack.config.CONFIG
        )
```

in both code paths, and filter afterwards.

Reply:
Done.